### ISSUE TITLE
CompilerException java.lang.ClassNotFoundException when using friend together with clj-httpclient

### ISSUE THREAD

**Issue by egli** [cemerick/friend#128]
Friend depends on an old version of `org.apache.httpcomponents/httpclient`. `clj-http` depends on a much newer version of the same. If you use both `friend` and `clj-http` in a project and `friend` happens to be before `clj-http` in the dependency vector `lein` will fetch the old httpclient. This will then cause an exception

```ShellSession
$ lein repl
nREPL server started on port 44499 on host 127.0.0.1 - nrepl://127.0.0.1:44499
REPL-y 0.3.5, nREPL 0.2.6
Clojure 1.6.0
OpenJDK 64-Bit Server VM 1.7.0_65-b32
Docs: (doc function-name-here)
(find-doc "part-of-name-here")
Source: (source function-name-here)
Javadoc: (javadoc java-object-or-class-here)
Exit: Control+D or (exit) or (quit)
Results: Stored in vars *1, *2, *3, an exception in *e
 
friend-and-clj-http1.core=> (require '[clj-http.client :as client])
 
CompilerException java.lang.ClassNotFoundException: org.apache.http.conn.ssl.SSLContexts, compiling:(clj_http/conn_mgr.clj:1:1)
```

The export from `lein deps :tree` can be found in two gists. The [first one](https://gist.github.com/egli/b3e149aded64c7628f8e) shows the dependencies if friend comes first and the [second one](https://gist.github.com/egli/4fa13bc791e52061f9e9) lists them if `clj-http` is first in the dependency vector.

 In a way this bug is related to #116 and in fact #124 should solve it.

**Comment by cemerick** [comment]
Fixed via #124, you can use `[com.cemerick/friend "0.2.2-SNAPSHOT"]` until I cut a final release.

**Comment by skelter** [comment]
I ran into this as well.  Looking forward to next release.
